homeassistant/components/abode/alarm_control_panel.py

```diff
@@ -74,3 +73,8 @@ def device_state_attributes(self):
             "battery_backup": self._device.battery,
             "cellular_backup": self._device.is_cellular,
         }
+
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this device."""
+        return f"{self._device.name} {self._device.device_id}"
```

Comment:
The alarm control panel does not have a `uuid` in Abode, only the devices paired to it. I looked through all of the data that is passed in the JSON for the alarm control panel and there really isn't a good option like `uuid`. The `id` assigned to the alarm control panel is `area_1`, which isn't very unique so I added the `name` property which gives it a `unique_id` in Home Assistant of: `Abode Alarm area_1`.

It's not ideal as I'd prefer to have something closer to `uuid` like the rest of the devices have. However, this should suffice in terms of being unique to any other integrations that might be added to Home Assistant (I can't imagine anything outside of Abode being assigned `Abode Alarm area_1`.

That being said, if we ever add support to have multiple Abode integrations into a single HA installation, then this `unique_id` would not work as they would conflict with each other.

Reply:
The unique_id only needs to be unique within the integration domain and the platform domain. So it needs to differentiate between abode alarm_control_panel entities.

We shouldn't add a unique id here if there is no unique stable identifier.

Reply:
Ah, okay. It just seems odd not having a unique id since every other device will show up in the entity registry except the alarm control panel.

Reply:
Figured out a solution. The mac address of the Abode hub (alarm) is contained within the login JSON data. I've submitted a [pull request](https://github.com/MisterWil/abodepy/pull/50) for abodepy that allows the mac address to be accessed from the `alarm` module. I spoke with @MisterWil and he will hopefully be able to merge the PR tomorrow and bump the abodepy version. Once that's done, I'll commit a small addition to this `alarm_control_panel.py` module:

```
    @property
    def unique_id(self):
        """Return a unique ID to use for this device."""
        return self._device.mac_address
```

Hopefully that should be it!

Reply:
Okay, @MisterWil did some great work such that the abode alarm (hub) will return its mac address when the same `self._device.device_uuid` is called so overriding `def unique_id(self):` in the alarm_control_panel.py is not necessary. Just pushed a commit that bumps abodepy to 0.16.3 which includes these changes.